arch/arm/src/common/arm_vfork.c

```diff
@@ -137,10 +137,19 @@ pid_t up_vfork(const struct vfork_s *context)
    * effort is overkill.
    */
 
-  newtop = STACK_ALIGN_DOWN((uint32_t)child->cmn.stack_base_ptr +
-                                      child->cmn.adj_stack_size -
-                                      XCPTCONTEXT_SIZE);
-  newsp = newtop - stackutil;
+  newsp = (uint32_t)child->cmn.stack_base_ptr +
+                    child->cmn.adj_stack_size - stackutil;
+
+  newtop = newsp - XCPTCONTEXT_SIZE;
+
+  /* Move the register context to newtop. */
+
+  memcpy((void *)newtop, child->cmn.stack_base_ptr +
+                         child->cmn.adj_stack_size -
+                         XCPTCONTEXT_SIZE, XCPTCONTEXT_SIZE);
```

Comment:
"child->cmn.stack_base_ptr + child->cmn.adj_stack_size - XCPTCONTEXT_SIZE" to tcb->xcpt->regs

Reply:
Done